core/jvm/src/main/scala/zio/sql/typetag.scala

```diff
@@ -34,7 +34,9 @@ trait TypeTagModule {
     implicit case object TString                                                  extends NotNull[String]
     implicit case object TUUID                                                    extends NotNull[UUID]
     implicit case object TZonedDateTime                                           extends NotNull[ZonedDateTime]
-    sealed case class TDialectSpecific[+A](typeTagExtension: TypeTagExtension[A]) extends NotNull[A]
+    sealed case class TDialectSpecific[+A](typeTagExtension: TypeTagExtension[A]) extends NotNull[A] {
+      def decode[A]: Either[Int, String] => A = ???
```

Comment:
I'd hazard a guess that this is your problem. I'm not sure exactly what you're trying to do here but keep in mind this has to work with all custom data types from any dialect.

Reply:
Ah yes I see what the issue is, will have a fix on this shortl!